src/cli/log.rs

```diff
@@ -152,25 +152,28 @@ where
 fn telemetry_default_tracer() -> Tracer {
     use std::time::Duration;
 
-    use opentelemetry::KeyValue;
+    use opentelemetry::{global, trace::TracerProvider, KeyValue};
     use opentelemetry_otlp::WithExportConfig;
     use opentelemetry_sdk::{
-        trace::{self, Sampler},
+        trace::{Config, Sampler},
         Resource,
     };
 
-    opentelemetry_otlp::new_pipeline()
+    let provider = opentelemetry_otlp::new_pipeline()
         .tracing()
         .with_exporter(
             opentelemetry_otlp::new_exporter()
                 .tonic()
                 .with_timeout(Duration::from_secs(3)),
         )
         .with_trace_config(
-            trace::config()
+            Config::default()
                 .with_sampler(Sampler::AlwaysOn)
                 .with_resource(Resource::new(vec![KeyValue::new("service.name", "rustup")])),
         )
         .install_batch(opentelemetry_sdk::runtime::Tokio)
-        .expect("error installing `OtlpTracePipeline` in the current `tokio` runtime")
+        .expect("error installing `OtlpTracePipeline` in the current `tokio` runtime");
+
+    global::set_tracer_provider(provider.clone());
```

Comment:
May I ask what this line is doing?

Reply:
This was previously part of the code that was there before, but upstream no longer implicitly does this:

https://github.com/open-telemetry/opentelemetry-rust/issues/1592
https://github.com/open-telemetry/opentelemetry-rust/issues/1810